### PR TITLE
Deploy Automation: Automate doing a release build.

### LINKED ISSUE #1100: Deploy Automation: Command to automate the release build
Related to #1096

We need a command to automate doing a build for release:

1. `git checkout master`
1. `git pull origin master`
1. `npm i`
1. `npm run build`
1. `rm -rf node_modules so these don't get packaged up in the release`

We also need to add a bunch of verification and error checking:

1. Display a message that shows the current version that we're building and asks the user to confirm before performing the build.
1. Need to make sure there are no uncommitted changes in the working copy.
1. _What other error checking can/should we do here?_
1. Verify that `npm run build` finished successfully.
  1. We can check the return code of the command.
  1. We can also verify that the files in the build directory exist.

### PR DESCRIPTION
**Pull request by markbiek**
### Changes proposed in this Pull Request:

This adds an initial command to checkout the latest `master` branch and do an `npm run build`. There is some rudimentary checking to ensure the build completed properly.

Closes #1100
Related to https://github.com/Automattic/wc-calypso-bridge/issues/1096

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Clone this branch.
2. In the `./scripts` directory, run `yarn` to manually install the script dependencies.

There are a number of different scenarios to check:

### Basic canceling of the script
1. In the project root, run `yarn create-release`.
1. At the initial confirmation prompt, type 'N' and verify the script stops running.
image

### Incorrect node version
1. Switch to a NodeJS version < v16.
1. In the project root, run `yarn create-release`.
1. You should get an `ERROR: Your version of NodeJS is not correct. Please install NodeJS v16.` error and the script should stop.

### Unclean working copy
1. Now make any sort of change in the project directory (a simple `touch t` to create a new file is fine).
1. In the project root, run `yarn create-release` and type 'y' to continue.
1. You should get a `ERROR: Working copy is not clean. Please commit or stash your changes before releasing.` error and the script should stop.

### Dev build is present
1. In the project root, run `npm i && npm start`. You can CTRL-C to cancel the `npm start` once it's done compiling.
1. In the project root, run `yarn create-release` and type 'y' to continue.
1. You should get an `ERROR: You may have a dev build. Please make sure you aren't running 'npm start'` error and the script should stop.

### Successful build
1. In the project root, run `rm -rf ./build` to clear out the dev build files.
1. In the project root, run `yarn create-release` and type 'y' to continue.
1. The script should run until you see a `SUCCESS: Release build complete` message.
1. Verify that the project root has a `build/` directory without any `.map` or empty files.
1. Verify that the `node_modules/` directory in the project root has been removed.
1. Verify that you are on the `master` branch.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by markbiek** [comment]
@valterlorran @daledupreez I would love any thoughts you have on additional types of error/safety checking we can do here.

File: scripts/commands/build.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		return false;
+	}
+
+	await pullLatestChanges();
+	const version = getCurrentVersion();
+	info(
+		`Pulled latest changes from 'master'. Current version is ${ version }.`
+	);
+
+	await execAsync( 'npm i' );
```
Comment (andres-blanco) [inline scripts/commands/build.js:53]: maybe it's useful to also check the node version based on [this file](https://github.com/Automattic/wc-calypso-bridge/blob/master/.nvmrc)?

Comment (markbiek) [reply scripts/commands/build.js:53]: Excellent idea, thank you!

**Comment by markbiek** [comment]
@andres-blanco I've added a check for the correct node version and added a test scenario to verify it. Thanks again for the suggestion.

**Review by andres-blanco — APPROVED** [review]
The code looks good and I went through the testing instructions successfully.

**Pull request merged**